stl/inc/xstring

```diff
@@ -2546,69 +2660,207 @@ public:
+
+#if _HAS_CXX20
+        if (_STD is_constant_evaluated()) { // Begin the lifetimes of the objects before copying to avoid UB
+            _Traits::assign(_Unfancy(_New_ptr), _New_capacity + 1, _Elem());
+        }
+#endif // _HAS_CXX20
+
+        _My_data._Mysize = _Count;
+        _My_data._Myres  = _New_capacity;
+        if constexpr (_Strat == _Construct_strategy::_From_char) {
+            _Traits::assign(_Unfancy(_New_ptr), _Count, _Arg);
+            _Traits::assign(_Unfancy(_New_ptr)[_Count], _Elem());
+        } else if constexpr (_Strat == _Construct_strategy::_From_ptr) {
+            _Traits::copy(_Unfancy(_New_ptr), _Arg, _Count);
+            _Traits::assign(_Unfancy(_New_ptr)[_Count], _Elem());
+        } else { // _Strat == _Construct_strategy::_From_string
+            _Traits::copy(_Unfancy(_New_ptr), _Arg, _Count + 1);
+        }
+
+        _ASAN_STRING_CREATE(*this);
         _Proxy._Release();
     }
 
+    template <class _Iter>
+    _CONSTEXPR20 void _Construct_from_iter(_Iter _First, const _Iter _Last) {
+        auto& _My_data  = _Mypair._Myval2;
+        auto& _Al       = _Getal();
+        auto&& _Alproxy = _GET_PROXY_ALLOCATOR(_Alty, _Al);
+        _Container_proxy_ptr<_Alty> _Proxy(_Alproxy, _My_data);
+
+        _My_data._Mysize = 0;
+        _My_data._Myres  = _BUF_SIZE - 1;
+#if _HAS_CXX20
+        if (_STD is_constant_evaluated()) {
+            _Construct_in_place(_My_data._Bx);
+            _My_data._Myres = _BUF_SIZE; // TRANSITION: constexpr SSO
+        }
+#endif // _HAS_CXX20
+
+        if constexpr (_Is_fwd_iter_v<_Iter>) {
+            const auto _Count = _Convert_size<size_type>(static_cast<size_t>(_STD distance(_First, _Last)));
+            if (_Count > max_size()) {
+                _Xlen_string(); // result too long
+            }
+
+#if _HAS_CXX20
+            const bool _Become_large = _Count >= _BUF_SIZE || _STD is_constant_evaluated();
+#else // ^^^ _HAS_CXX20 / !_HAS_CXX20 vvv
+            const bool _Become_large = _Count >= _BUF_SIZE;
+#endif // _HAS_CXX20
+            if (_Become_large) {
+                const size_type _New_capacity = _Calculate_growth(_Count);
+                const pointer _New_ptr        = _Al.allocate(_New_capacity + 1); // throws
+                _Construct_in_place(_My_data._Bx._Ptr, _New_ptr);
+                _My_data._Myres = _New_capacity;
+
+#if _HAS_CXX20
+                if (_STD is_constant_evaluated()) { // Begin the lifetimes of the objects before copying to avoid UB
+                    _Traits::assign(_Unfancy(_New_ptr), _New_capacity + 1, _Elem());
+                }
+#endif // _HAS_CXX20
+            }
+        }
+
+        _Tidy_deallocate_guard<basic_string> _Guard{this};
+        for (; _First != _Last; ++_First) {
+            if constexpr (!_Is_fwd_iter_v<_Iter>) {
+                if (_My_data._Mysize >= _My_data._Myres) { // Need to grow
+                    if (_My_data._Mysize > max_size()) {

```

Comment:
How could `_Mysize` ever be strictly greater than `max_size()`, that should be forbidden by invariants?

Reply:
Dito erroneuos inversion of the condition. Should be `_My_data._Mysize == max_size()`